### PR TITLE
Add support for the ping event to info-typer

### PR DESCRIPTION
**Pull request by johnworth**
I'm adding support for the ping event to every AMQP service, so now it's info-typer's turn. However, info-typer was only connecting to the iRODS AMQP broker and exchange on the default vhost and the code wasn't flexible enough to allow us to simultaneously connect to another broker/vhost and exchange without duplicating a bunch of code.

The info-typer.events namespace is where the new ping event handling code has been added.

I've pulled a lot of direct configuration accesses out of (configure) and put them into task-specific configuration maps that get passed in as parameters.

I've added support for URI-based connection configurations since it reduces the number of configuration settings that must be managed for each AMQP connection.

A corresponding pull request against the cyverse-de/ansible repo will be opened soon.

### CODE REVIEW

File: src/info_typer/config.clj [hunk 1]
```diff
@@ -97,28 +97,46 @@
   "info-typer.irods.use-trash" true)
 
 
-(cc/defprop-optstr amqp-host
-  "The hostname for the AMQP broker"
+(cc/defprop-optstr amqp-events-uri
+  "The hostname for the events AMQP broker"
   [props config-valid configs]
-  "info-typer.amqp.host" "rabbit")
+  "info-typer.events.amqp.uri" "")
```
Comment (strootman) [inline src/info_typer/config.clj]: This should have some default. `amqp://guest:guest@rabbit:5672`. Requesting change.

File: src/info_typer/events.clj [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  (:import [org.cyverse.events.ping PingMessages
+                                    PingMessages$Ping
+                                    PingMessages$Pong]
+           [com.google.protobuf.util JsonFormat]))
+
+(def pong-key "events.info-typer.pong")
+
+(defn ping-handler
+  [channel metadata payload]
+  (lb/publish channel (cfg/amqp-events-exchange) pong-key
```
Comment (strootman) [inline src/info_typer/events.clj:22]: When publishing, we should create a header which identifies the publisher. In this case, the service. This is to help with debugging in the future, in the case that a particular topic/routing-key is published from two different services.

Comment (strootman) [reply src/info_typer/events.clj:22]: Furthermore, this header will be parseable from ES, allowing us to filter traffic from specified producer(s).

File: src/info_typer/config.clj [hunk 3]
```diff
@@ -97,28 +97,46 @@
   "info-typer.irods.use-trash" true)
 
 
-(cc/defprop-optstr amqp-host
```
Comment (strootman) [inline src/info_typer/config.clj:100]: Tests are currently failing because of the changes to this file.

**Review by strootman — CHANGES_REQUESTED** [review]
Overall, it looks good. 
One of the comments is a change request, and is marked as such.
The other comments should be considered as comments towards future work/design-discussions.

File: src/info_typer/config.clj (hunk 3 above)
Comment (johnworth) [reply src/info_typer/config.clj:100]: This should be fixed.

File: src/info_typer/config.clj (hunk 1 above)
Comment (johnworth) [reply src/info_typer/config.clj]: This should be fixed.

**Review by strootman — APPROVED** [review]

**Pull request merged**